### PR TITLE
New sync edge cases

### PR DESCRIPTION
**Pull request by czue**
http://manage.dimagi.com/default.asp?179664

@dannyroberts I think this is the primary source of the sync log rebuilds we found.

i'm not bothering to fix it on the old version for now since we believe we migrated most of the problem domains.

cc @snopoke

### CODE REVIEW

**Comment by czue** [comment]
(wait for tests)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 1]
```diff
@@ -570,7 +571,18 @@ def _remove_case(to_remove):
...
-            self.case_ids_on_phone.remove(to_remove)
+            try:
+                self.case_ids_on_phone.remove(to_remove)
+            except KeyError:
+                # todo: this here to avoid having to manually clean up after
+                # http://manage.dimagi.com/default.asp?179664
+                # it should be removed when there are no longer any instances of the assertion
+                if self.date < datetime(2015, 8, 25):
+                    _assert = soft_assert(to=['czue' + '@' + 'dimagi.com'], exponential_backoff=False)
+                    _assert(False, 'patching sync log {} to remove missing case ID {}!'.format(self._id, to_remove))
```
Comment (dimagimon) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: E501 line too long (116 > 115 characters)

**Comment by benrudolph** [comment]
:+1:

**Comment by dannyroberts** [comment]
This is awesome to see you getting through the new-sync backlog of issues! Can't wait for the big delete :)

File: corehq/ex-submodules/casexml/apps/phone/models.py [hunk 2]
```diff
@@ -627,6 +641,11 @@ def update_phone_lists(self, xform, case_list):
                         to_prune.add(case._id)
                         made_changes = True
                     else:
+                        if phone_owns_case and not log_has_case:
+                            # this can happen if a create sets the owner id to something invalid
+                            # and an update in the same block/form sets it back to valid
+                            self._add_primary_case(case._id)
+                            made_changes = True
```
Comment (snopoke) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: Ha, that's a tricky one.

Comment (czue) [inline corehq/ex-submodules/casexml/apps/phone/models.py]: tell me about it.

**Pull request merged**